src/primitive_file_win.cc

```diff
@@ -405,111 +403,104 @@ PRIMITIVE(stat) {
 }
 
 PRIMITIVE(unlink) {
-  ARGS(cstring, pathname);
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
+  ARGS(StringOrSlice, pathname);
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  int result = unlink(path);
+  int result = _wunlink(path);
   if (result < 0) return return_open_error(process, errno);
   return process->program()->null_object();
 }
 
 PRIMITIVE(rmdir) {
-  ARGS(cstring, pathname);
-  char path[MAX_PATH];
-  auto error = get_absolute_path(process, pathname, path);
-  if (error) return error;
+  ARGS(StringOrSlice, pathname);
+  BLOB_TO_ABSOLUTE_PATH(path, pathname);
 
-  if (RemoveDirectory(path) == 0) WINDOWS_ERROR;
+  if (RemoveDirectoryW(path) == 0) WINDOWS_ERROR;
   return process->program()->null_object();
 }
 
 PRIMITIVE(rename) {
-  ARGS(cstring, old_name, cstring, new_name);
-  int result = rename(old_name, new_name);
+  ARGS(StringOrSlice, old_name_blob, StringOrSlice, new_name_blob);
+  WideCharAllocationManager allocation1(process), allocation2(process);
+  wchar_t* old_name = allocation1.to_wcs(&old_name_blob);
+  wchar_t* new_name = allocation2.to_wcs(&new_name_blob);
+  // TODO(erik): Should this use absolute path?
```

Comment:
Yes.
Since we change current working directory without the system knowing about it, we can't give the system relative paths.
Add a test, where we change directory and then move a file.